### PR TITLE
Add Application.wrapSync()

### PR DESCRIPTION
**Pull request by aziz512**
@pbaize @rdepena

### CODE REVIEW

File: src/api/application/application.ts [hunk 1]
```diff
@@ -42,13 +42,23 @@ export default class ApplicationModule extends Base {
...
-     * @return {Promise.<Application>}
+     * @return Promise<Application>
      * @tutorial Application.wrap
      */
     public wrap(identity: Identity): Promise<Application> {
         return Promise.resolve(new Application(this.wire, identity));
     }
 
+    /**
+     * Returns an Application object that represents an existing application.
```
Comment (tgoc99) [inline src/api/application/application.ts]: Maybe mention 'synchronous' somewhere in the description?  (and add 'asynchronous' to other one)

File: src/api/application/application.ts [hunk 2]
```diff
@@ -42,13 +42,23 @@ export default class ApplicationModule extends Base {
...
+     * @param { Identity } indentity
+     * @return Application
+     * @tutorial Application.wrapSync
```
Comment (tgoc99) [inline src/api/application/application.ts]: Need to add tutorial in tutorials folder.

File: src/api/frame/frame.ts [hunk 3]
```diff
@@ -23,13 +23,31 @@ export default class _FrameModule extends Base {
...
     /**
      * Get a reference to the current frame
      * @return {Promise.<_Frame>}
      */
     public getCurrent(): Promise<_Frame> {
         return Promise.resolve(new _Frame(this.wire, this.me));
     }
+
+    /**
+     * Synchronously get a reference to the current frame
```
Comment (pbaize) [inline src/api/frame/frame.ts]: "returns" rather than "get"

File: src/api/application/application.ts [hunk 4]
```diff
@@ -40,15 +40,25 @@ export interface TrayInfo {
 
 export default class ApplicationModule extends Base {
     /**
-     * Returns an Application object that represents an existing application.
+     * Asynchronously returns an Application object that represents an existing application.
      * @param { Identity } indentity
-     * @return {Promise.<Application>}
+     * @return Promise<Application>
```
Comment (licui3936) [inline src/api/application/application.ts]: original one  is ok.

File: src/api/application/application.ts [hunk 5]
```diff
@@ -40,15 +40,25 @@ export interface TrayInfo {
...
      * @tutorial Application.wrap
      */
     public wrap(identity: Identity): Promise<Application> {
         return Promise.resolve(new Application(this.wire, identity));
     }
 
+    /**
+     * Synchronously returns an Application object that represents an existing application.
+     * @param { Identity } indentity
```
Comment (licui3936) [inline src/api/application/application.ts]: typo error

File: src/api/application/application.ts [hunk 6]
```diff
@@ -40,15 +40,25 @@ export interface TrayInfo {
...
+     * @return Application
+     * @tutorial Application.wrapSync
```
Comment (licui3936) [inline src/api/application/application.ts]: missing curly bracket for jsdoc @return.

File: src/api/application/application.ts [hunk 7]
```diff
@@ -69,6 +79,15 @@ export default class ApplicationModule extends Base {
         return this.wrap({ uuid: this.wire.me.uuid });
     }
 
+    /**
+     * Synchronously returns an Application object that represents the current application
+     * @return Application>
```
Comment (licui3936) [inline src/api/application/application.ts]: same here

File: src/api/external-application/external-application.ts [hunk 8]
```diff
@@ -8,13 +8,22 @@ export interface ExternalApplicationInfo {
...
      * @return {Promise.<ExternalApplication>}
      */
     public wrap(uuid: string): Promise<ExternalApplication> {
         return Promise.resolve(new ExternalApplication(this.wire, {uuid}));
     }
+
+    /**
+     * Synchronously returns an External Application object that represents an existing external application.
+     * @param {string} uuid The UUID of the external application to be wrapped
+     * @return ExternalApplication
```
Comment (licui3936) [inline src/api/external-application/external-application.ts]: same here

File: src/api/frame/frame.ts [hunk 9]
```diff
@@ -24,12 +24,30 @@ export default class _FrameModule extends Base {
     }
 
     /**
-     * Get a reference to the current frame
+     * Synchronously returns a reference to the specified frame. The frame does not have to exist
+     * @param {string} uuid - uuid of the frame you want to wrap
+     * @param {string} name - name of the frame you want to wrap
+     * @return _Frame
+     */
```
Comment (licui3936) [inline src/api/frame/frame.ts]: same here

File: src/api/frame/frame.ts [hunk 10]
```diff
@@ -24,12 +24,30 @@ export default class _FrameModule extends Base {
...
+    public wrapSync(uuid: string, name: string): _Frame {
+        return new _Frame(this.wire, {uuid, name});
+    }
+
+    /**
+     * Returns a reference to the current frame
```
Comment (licui3936) [inline src/api/frame/frame.ts]: missing "Asynchronously "

File: src/api/frame/frame.ts [hunk 11]
```diff
@@ -24,12 +24,30 @@ export default class _FrameModule extends Base {
...
      * @return {Promise.<_Frame>}
      */
     public getCurrent(): Promise<_Frame> {
         return Promise.resolve(new _Frame(this.wire, this.me));
     }
+
+    /**
+     * Synchronously returns a reference to the current frame
+     * @return _Frame
```
Comment (licui3936) [inline src/api/frame/frame.ts]: add curly bracket around _Frame

File: src/api/window/window.ts [hunk 12]
```diff
@@ -12,9 +12,9 @@ export default class _WindowModule extends Base {
     private instance: _Window;
 
     /**
-     * Returns a Window object that represents an existing window.
+     * Asynchronously returns a Window object that represents an existing window.
      * @param { Identity } indentity
-     * @return {Promise.<_Window>}
+     * @return Promise<_Window>
```
Comment (licui3936) [inline src/api/window/window.ts]: keep the original one

File: src/api/window/window.ts [hunk 13]
```diff
@@ -24,6 +24,19 @@ export default class _WindowModule extends Base {
         return Promise.resolve(this.instance);
     }
 
+    /**
+     * Synchronously returns a Window object that represents an existing window.
+     * @param { Identity } indentity
+     * @return _Window
```
Comment (licui3936) [inline src/api/window/window.ts]: add bracket

File: src/api/window/window.ts [hunk 14]
```diff
@@ -43,6 +56,15 @@ export default class _WindowModule extends Base {
     public getCurrent(): Promise<_Window> {
         return this.wrap(this.wire.me);
     }
+
+    /**
+     * Synchronously returns a Window object that represents the current window
+     * @return _Window
```
Comment (licui3936) [inline src/api/window/window.ts]: add bracket

Comment (licui3936) [reply src/api/window/window.ts]: Also missing 'Asynchronously' for getCurrent function

File: src/api/frame/frame.ts [hunk 15]
```diff
@@ -24,12 +24,30 @@ export default class _FrameModule extends Base {
     }
 
     /**
-     * Get a reference to the current frame
+     * Synchronously returns a reference to the specified frame. The frame does not have to exist
+     * @param {string} uuid - uuid of the frame you want to wrap
+     * @param {string} name - name of the frame you want to wrap
+     * @return {_Frame}
+     */
+    public wrapSync(uuid: string, name: string): _Frame {
```
Comment (tgoc99) [inline src/api/frame/frame.ts]: We should make this match Window in taht it should take identity as one argument as an obj (done incorrectly originally, should fix that too.)

File: src/api/frame/frame.ts [hunk 16]
```diff
@@ -24,12 +24,30 @@ export default class _FrameModule extends Base {
     }
```
Comment (tgoc99) [inline src/api/frame/frame.ts]: Make wrap for Frame look like Window (take identity object)

Comment (tgoc99) [reply src/api/frame/frame.ts]: May need to fix tutorial for frame wrap as well.

**Pull request merged**